applications/klighd-vscode/src/klighd-webview-reopener.ts

```diff
@@ -22,20 +22,31 @@ import { StorageService } from "./storage/storage-service";
 export class KlighdWebviewReopener {
 
     private readonly storage: StorageService
+    private toDispose: { dispose(): any }[] = []
 
     constructor(storage: StorageService) {
         this.storage = storage
     }
 
     onExtensionCreated(): void {
         const diagramWasOpen = this.storage.getItem('diagramOpen')
-        if (diagramWasOpen == undefined || diagramWasOpen) {
-            const uri = window.activeTextEditor?.document.fileName
-            if (uri) {
+        if (diagramWasOpen === undefined || diagramWasOpen) {
+            const activeTextEditor = window.activeTextEditor
+            if (activeTextEditor) {
+                const uri = activeTextEditor.document.fileName
                 commands.executeCommand(command.diagramOpen, Uri.file(uri))
+            } else {
+                // Register this an active editor changed to open the diagram then.
+                this.toDispose.push(window.onDidChangeActiveTextEditor(editor => {
+                    let uri = undefined
+                    if (editor) {
+                        uri = editor.document.uri
+                    }
+                    commands.executeCommand(command.diagramOpen, uri)
```

Comment:
So you call the diagramOpen command with `undefined` if the active editor changes to `undefined`? What does that even do?

> *Note* that the event also fires when the active editor changes to `undefined`.

Reply:
I don't care and this will not happen since it cannot be called if the editor is undefined.